# Quiz – Python Calculations and Loops

## Question 1  
Use Python to calculate the expression:

3 × 8^33 + 7^5

In [2]:
3 * 8**33 + 7**5

1901475900342344102245054824871

### Explanation

- `**` is the exponent operator in Python.
- `8**33` calculates 8 raised to the 33rd power.
- `7**5` calculates 7 raised to the 5th power.
- Python handles very large integers automatically.

## Question 2  
Determine the data type of:

1 + 2  
1 + 2.

In [3]:
type(1 + 2)

int

In [4]:
type(1 + 2.)

float

### Explanation

- `1 + 2` returns an integer (`int`) because both numbers are integers.
- `1 + 2.` returns a float (`float`) because `2.` contains a decimal.
- In Python, if at least one number is a float, the result becomes a float.

### Explanation

- `1 + 2` returns an integer (`int`) because both numbers are integers.
- `1 + 2.` returns a float (`float`) because `2.` contains a decimal.
- In Python, if at least one number is a float, the result becomes a float.

In [5]:
for n in range(1, 6):
    print(n**2)

1
4
9
16
25


### Explanation

- `range(1, 6)` generates numbers from 1 to 5.
- `n**2` calculates n squared.

## Question 4  

Create the variable `i` and assign it the value 1.  
Create a loop that prints `i²`.  
Increment `i` by 1 until `i²` exceeds 1000.  
Print the final value of `i`.

In [6]:
i = 1

while i**2 <= 1000:
    print(i**2)
    i += 1

print("Final value of i:", i)

1
4
9
16
25
36
49
64
81
100
121
144
169
196
225
256
289
324
361
400
441
484
529
576
625
676
729
784
841
900
961
Final value of i: 32


### Explanation

The loop continues while:

i² ≤ 1000

31² = 961 (still less than 1000)  
32² = 1024 (greater than 1000)

Therefore, the loop stops when i = 32.

Final answer: i = 32